example/lib/main.dart

```diff
@@ -95,150 +74,119 @@ class _MyAppState extends State<MyApp> {
-                      "3",
-                      simplePeriodicTask,
-                      initialDelay: Duration(seconds: 10),
+                    Workmanager().registerOneOffTask(
+                      rescheduledTaskKey,
+                      rescheduledTaskKey,
+                      inputData: <String, dynamic>{
+                        'key': Random().nextInt(64000),
+                      },
                     );
-                  }),
-              //This task runs periodically
-              //It will run about every hour
-              PlatformEnabledButton(
-                  platform: _Platform.android,
-                  child: Text("Register 1 hour Periodic Task"),
+                  },
+                ),
+                ElevatedButton(
+                  child: Text("Register failed Task"),
                   onPressed: () {
-                    Workmanager().registerPeriodicTask(
-                      "5",
-                      simplePeriodic1HourTask,
-                      frequency: Duration(hours: 1),
+                    Workmanager().registerOneOffTask(
+                      failedTaskKey,
+                      failedTaskKey,
                     );
-                  }),
-              SizedBox(height: 16),
-              Text(
-                "Task cancellation",
-                style: Theme.of(context).textTheme.headline5,
-              ),
-              ElevatedButton(
-                child: Text("Cancel All"),
-                onPressed: () async {
-                  await Workmanager().cancelAll();
-                  print('Cancel all tasks completed');
-                },
-              ),
-            ],
+                  },
+                ),
+                //This task runs once
+                //This wait at least 10 seconds before running
+                ElevatedButton(
+                    child: Text("Register Delayed OneOff Task"),
+                    onPressed: () {
+                      Workmanager().registerOneOffTask(
+                        simpleDelayedTask,
+                        simpleDelayedTask,
+                        initialDelay: Duration(seconds: 10),
+                      );
+                    }),
+                SizedBox(height: 8),
+                //This task runs periodically
+                //It will wait at least 10 seconds before its first launch
+                //Since we have not provided a frequency it will be the default 15 minutes
+                ElevatedButton(
+                    child: Text("Register Periodic Task (Android)"),
+                    onPressed: Platform.isAndroid
+                        ? () {
+                            Workmanager().registerPeriodicTask(
+                              simplePeriodicTask,
+                              simplePeriodicTask,
+                              initialDelay: Duration(seconds: 10),
+                            );
+                          }
+                        : null),
```

Comment:
This is slightly different behavior than before. Future versions of the example will have a dialog to explain the situation about periodic tasks to the user.

Reply:
@ened Should I do anything with this?